00001-enarx-sprint-framework/README.md

```diff
@@ -0,0 +1,542 @@
+
+![PR workflow diagram](./pr-workflow.svg)
+
+As with every issue, every PR will be reviewed through a Triage, during which
+reviewers will be requested and assigned. At least two reviewers must be
+assigned for review, though the PR author may optionally request other people
+review it without explicitly assigning them. These "optional reviewers" may
+choose not to review it; however, if they _do_ submit a review, they are
+expected to follow through the entire process until the PR is merged.
+
+Note that the flowchart follows the flow for an individual reviewer and the
+author; this process is replicated for every individual reviewer involved with
+the PR.
+
+The cycle starts with the `reviewer_assigned` state, which indicates that it's
+the reviewer's responsibility to submit a PR review.
+
+Once the reviewer submits a review, two things can happen based on the content:
+
+- If the review approves of the PR, it moves into the `approved` state for that
+  individual author. From here, the PR waits on other reviewers until it meets
+  the criteria for merge (detailed [below](#merge-criteria)).
+- If the review requests changes, it moves into the `reviewed` state. A PR can
+  arrive here either through an assigned reviewer submitting a review, or
+  through a "drive-by" reviewer that requests changes.
+
+Once a PR is in the `reviewed` state, automation will take care of updating
+assignees to reflect whose responsibility the PR is at the present time:
+
+- The reviewer that submitted the review is unassigned.
+- The PR author is assigned to the PR to address the reviewer's feedback.
+
+Once these two are done, the PR moves into the `author_assigned` state. At this
+point, it's the author's responsibility to address the review they've received.
+
+When the author believes they have addressed the reviewer's feedback, they must
+request a re-review using the button present in the Assignees tab next to the
+Changes Requested indicator:
+
+![PR workflow diagram](request-rereview.png)
+
+This is represented by the `request_rereview` state
+in the flowchart. From here, automation again helps the process along:
+
+- If the author does not have any other outstanding reviews, they are unassigned
+  from the PR.
+- The reviewer that submitted the now-addressed review is re-assigned, and the
+  PR is moved back to the `reviewer_assigned` state.
+  
+At this point, the cycle is complete, and it repeats from the beginning until
+the reviewer is satisfied with the PR and submits their approval.
+
+### Merge criteria
+
+With a clear framework in place for reviewing and approving pull requests, it's
+now more feasible for the Enarx project to begin enforcing stricter merge
+requirements.
+
+A pull request should only be merged when:
+
+- it has two or more approving reviews
+- every reviewer that has submitted a review has updated their review to
+  "Approved"
+
+The latter exists to ensure that all feedback from all reviewers is addressed
```

Comment:
Do we want to require tests for code-based PRs?

Reply:
I suppose I assumed that was implied. Code tests are currently required to pass anyway. We can make that an explicit requirement for merge if desired. (Makes more sense thinking about it, especially in the context of potentialy auto-merging some PRs.)